# Sentiment Analyses

#### Sentiment Analyses uses *Polarity* and *Subjectivity* to analyze text

## Polarity
- Measures the sentiment or emotion of a text, ranging from -1 to 1 (Negative and Positive)

## Subjectivity
- Indicates how much of the text is based on personal opinion, ranging from 0 to 1 (Objective and Subjective)

## Downloads

ite140 % pip3 install nbformat
ite140 % pip3 install spacytextblob
ite140 % pip3 install spacy
ite140 % pip3 install textblob
ite140 % pip3 install spacy download en_core_web_sm
ite140 % pip3 install https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.5.0/en_core_web_sm-3.5.0.tar.gz


In [32]:
import spacy
from spacytextblob.spacytextblob import SpacyTextBlob


In [33]:
nlp = spacy.load('en_core_web_sm')
nlp.add_pipe('spacytextblob')

#Example 1

text = 'I had a really horrible day. It was the worst day ever! But every now and then I have a really good day that makes me happy.'
doc = nlp(text)

/Users/1003427/Library/Python/3.9/lib/python/site-packages/spacy/util.py:910: UserWarning: [W095] Model 'en_core_web_sm' (3.5.0) was trained with spaCy v3.5.0 and may not be 100% compatible with the current version (3.8.2). If you see errors or degraded performance, download a newer compatible model or retrain your custom model with the current spaCy version. For more details and available updates, run: python -m spacy validate
  warnings.warn(warn_msg)


In [34]:
doc._.blob.polarity

-0.125

A *polarity* of -0.125 indicates that the statement has an almost equal amount of positive and negative tone, but is slightly more negative

In [35]:
doc._.blob.subjectivity

0.9

A *Subjectivity* of 0.9 indicates that the text is mostly made up of words that have strong emotions

Lets do another example

In [36]:
# Example 2
text = 'The product did not meet the advertised specifications, with defects in build quality and multiple malfunctions occuring within one week of use'
doc = nlp(text)

In [37]:
doc._.blob.polarity

0.0

A *Polarity* of-0 indicates the statement has no emotional words

In [38]:
doc._.blob.subjectivity

0.0

A *subjectivity* of 0 indicates that the statement is solely based on facts and no personal feelings or judgements whatsoever

## Summary 
- Sentiment Anaylses uses polarity and subjectivity to evaluate a statement based on its emotional words, and bias from the author